stmhal/sdcard.c

```diff
@@ -76,6 +76,7 @@
 // TODO: I think that as an optimization, we can allocate these dynamically
 //       if an sd card is detected. This will save approx 260 bytes of RAM
 //       when no sdcard was being used.
+static bool sdcard_present = false;
```

Comment:
For the case with no mechanical switch, but where we can read the sdcard presence before initialising the SDIO, I wanted `sdcard_is_present()` to return either true or false, based on the initial detection.

I'm not sure how that can be done without a variable to cache the initial detection state.